rest/oidc_test_provider.go

```diff
@@ -49,12 +49,25 @@ const (
 type TokenRequestType uint8
 
 const (
-	TokenRequest_AuthCode TokenRequestType = iota
-	TokenRequest_Refresh
+	TokenRequestAuthCode TokenRequestType = iota
+	TokenRequestRefresh
 )
 
 var testProviderAudiences = []string{testProviderAud} // Audiences in array format for test provider validation
 
+// identityTokenFormat represents a specific token format supported by an OpenID Connect provider.
+type identityTokenFormat string
+
+const (
+
+	// defaultFormat represents the common format of ID token supported by most of the providers.
```

Comment:
Modify this comment to list the providers that you verified are using the default approach.

Reply:
Sure Adam, updated this comment with the list of providers. Also added token simulation for Microsoft Identity Platform V2.0 format and Yahoo! token format.